templates/Makefile.template

```diff
@@ -1503,11 +1515,17 @@
   ifeq ($(SYSTEM),Darwin)
   	$(Q) ${ld} ${ldflags} -L$(LIBDIR)/$(CONFIG) -install_name $(SHARED_PREFIX)${lib.name}$(SHARED_VERSION).$(SHARED_EXT) -dynamiclib -o ${out_libbase}.$(SHARED_EXT) ${common}${libs}
   else
+  ifeq ($(SYSTEM),HP-UX)
+  	$(Q) ${ld} ${ldflags} -L$(LIBDIR)/$(CONFIG) -Wl,-E -Wl,+s -Wl,+b,/usr/local/lib -shared -o ${out_libbase}.$(SHARED_EXT) ${common}${libs}
```

Comment:
HPUX ld relevant flags:

- -E

   Mark all symbols defined by a program for export to shared libraries.

- +s

  Indicate that at run-time, the dynamic loader can use the
      environment variables SHLIB_PATH and LD_LIBRARY_PATH to locate shared libraries.

- +b path_list   
Specify a colon-separated list of directories to be searched at program run-time to locate shared libraries needed by the executable output file.